src/Runner.Worker/ActionManager.cs

```diff
@@ -613,14 +613,35 @@ private async Task DownloadRepositoryActionAsync(IExecutionContext executionCont
                 }
 #endif
 
-                // repository archive from github always contains a nested folder
+                // repository archive from github always contains a nested folder, the nested folder contains the short SHA for the ref
+                // ex: actions/checkout@master the nested folder looks like actions-checkout-01aeccc
+                // However, when you reference action using tag, the SHA in the folder name is not really the commit SHA, it's the SHA for the tag itself.
+                var shortSha = "";
                 var subDirectories = new DirectoryInfo(stagingDirectory).GetDirectories();
                 if (subDirectories.Length != 1)
                 {
                     throw new InvalidOperationException($"'{archiveFile}' contains '{subDirectories.Length}' directories");
                 }
                 else
```

Comment:
Not related to this PR, but is the style to actually have an `else` after `throw`/`return`?